src/main/java/com/cegeka/academy/web/rest/ChallengeController.java

```diff
@@ -31,6 +32,11 @@ public void delete(@PathVariable long id) throws NotFoundException {
         challengeService.deleteChallenge(id);
     }
 
+    @RequestMapping(value = "/{id}", method = RequestMethod.PUT)
+    public ChallengeDTO updateChallenge(@Valid @RequestBody ChallengeDTO challengeDTO, @PathVariable Long challengeId) throws NotFoundException {
```

Comment:
In requestMapping pathVariable se numeste id, iar in lista de param este challengeId, ar trebui ca in @PathVariable se specifici ca in challengeId doresti sa pui valoarea din "id", altfel o sa apara probleme.